x/ecocredit/server/msg_server.go

```diff
@@ -193,28 +242,26 @@ func (s serverImpl) CreateBatch(goCtx context.Context, req *ecocredit.MsgCreateB
 	amountCancelledStr := math.NewDecFromInt64(0).String()
 
 	err = s.batchInfoTable.Create(ctx, &ecocredit.BatchInfo{
-		ClassId:         classID,
+		ProjectId:       projectID,
 		BatchDenom:      string(batchDenom),
-		Issuer:          req.Issuer,
 		TotalAmount:     totalSupplyStr,
 		Metadata:        req.Metadata,
 		AmountCancelled: amountCancelledStr,
 		StartDate:       req.StartDate,
 		EndDate:         req.EndDate,
-		ProjectLocation: req.ProjectLocation,
 	})
 	if err != nil {
 		return nil, err
 	}
 
 	err = ctx.EventManager().EmitTypedEvent(&ecocredit.EventCreateBatch{
-		ClassId:         classID,
+		ProjectId:       projectID,
 		BatchDenom:      string(batchDenom),
 		Issuer:          req.Issuer,
 		TotalAmount:     totalSupplyStr,
 		StartDate:       req.StartDate.Format("2006-01-02"),
 		EndDate:         req.EndDate.Format("2006-01-02"),
-		ProjectLocation: req.ProjectLocation,
+		ProjectLocation: projectInfo.ProjectLocation,
```

Comment:
I think it does not hurt to include project location in the event. I think this is ok as is and removing it would be out of scope of this pull request.

Reply:
We want to keep the chain state normalized and reduce the information to minimum (we pay for every bit) if possible (unless it hurts UX). So, I'm not sure how much value it has to include it here. If we really need it then all good, if it's nice to have then probably we should reconsider. @clevinson , @aaronc what do you think?

Reply:
ping

Reply:
I'm a bit split on this. Until we separate state machine and client concerns better, it's sort of okay to pollute the state machine with stuff just needed for clients. In the future we may be able to have event serialization be optional and outside of tendermint. So I think it's okay, but not a big deal either way since credit issuance is not a high frequency event and polling is fine.